types/relay-runtime/lib/store/readInlineData.d.ts

```diff
@@ -0,0 +1,19 @@
+import { GraphQLTaggedNode } from '../query/RelayModernGraphQLTag';
+
+interface _RefType<T> {
+    ' $refType': T;
+}
+interface _FragmentRefs<T> {
+    ' $fragmentRefs': T;
+}
+
+type FragmentRef<T> = T extends _RefType<infer U> ? _FragmentRefs<U> : never;
```

Comment:
helper type to get something that holds onto the fragment ref (see `react-relay` test)

Reply:
i think we need to export some kind of helper like this to make it reasonably possible for users to type `readInlineData` calls

Reply:
Sure, let's export it from the top-level module?

Reply:
yup, it's exported from both

Reply:
But then this module could perhaps just import it from the top-level module?

Reply:
it doesn't look like we have any other types declared at top level in `relay-runtime`. do you think that makes sense to add / wouldn't lead to circular import issues?

Reply:
It shouldn’t lead to import issues afaik. I’d rather have them there then introduce a new file that doesn’t exist upstream or defining these trickier types in more than one place.